go/stats/prombackend/prombackend.go

```diff
@@ -0,0 +1,63 @@
+package prombackend
+
+import (
+	"net/http"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"vitess.io/vitess/go/stats"
+)
+
+// PromBackend implements PullBackend using Prometheus as the backing metrics storage
+type PromBackend struct {
+	namespace string
+}
+
+func InitPromBackend(namespace string) {
+	http.Handle("/metrics", promhttp.Handler())
+	promBackend := NewPromBackend(namespace)
+	stats.RegisterPullBackendImpl("prom", promBackend)
+}
+
+func NewPromBackend(namespace string) *PromBackend {
```

Comment:
This doesn't seem to me something that warrants another public API function... better IMO would be to just fold this in as part of Init.